dao/src/test/java/se/fortnox/reactivewizard/db/DbProxyTest.java

```diff
@@ -410,57 +478,48 @@ public void testAllPathsOfTryWithResourceUsingCount() throws SQLException {
         insertAndAssertError(insertCount(), "prepare statement");
     }
 
-    private Observable<Integer> insertCount() {
+    private Flux<Integer> insertCount() {
         return dbProxyTestDao.insertLocalDate(LocalDate.MAX);
     }
 
     private void insertAndAssertError(String message) {
         insertAndAssertError(dbProxyTestDao.insert(), message);
     }
 
-    private void insertAndAssertError(Observable<?> insert, String message) {
-        try {
-            insert.toBlocking().singleOrDefault(null);
-            fail("expected exception");
-        } catch (Exception e) {
-            Exception cause = (Exception) e.getCause();
-            if (cause != null) {
-                e = cause;
-            }
-            if (message != null) {
-                assertThat(e.getMessage()).contains(message);
-            }
-        }
+    private void insertMonoAndAssertError(String message) {
+        insertAndAssertError(dbProxyTestDao.insertMono().flux(), message);
     }
 
     private void insertAndAssertError(Flux<?> insert, String message) {
         assertThatExceptionOfType(Exception.class)
                 .isThrownBy(insert::blockLast)
                 .satisfies(exception -> {
                     Throwable selfOrCause = Optional.ofNullable(exception.getCause()).orElse(exception);
-                    assertThat(selfOrCause.getMessage()).contains(message);
+                    if (Objects.nonNull(message)) {
+                        assertThat(selfOrCause.getMessage()).contains(message);
+                    }
                 });
     }
 
     @Test
     public void shouldPassIfUpdateReturnsZeroAffectedRowsAndQueryAllowsZeroUpdates() throws SQLException {
         mockDb.setUpdatedRows(0);
         assertThatNoException()
-                .isThrownBy(() -> dbProxyTestDao.updateAllowingZeroAffectedRows("mykey", "myval").toBlocking().singleOrDefault(null));
+                .isThrownBy(() -> dbProxyTestDao.updateAllowingZeroAffectedRows("mykey", "myval").block());
     }
 
     @Test
     public void shouldAllowVoidReturnType() throws SQLException {
         mockDb.setUpdatedRows(1);
-        assertThat(dbProxyTestDao.insertVoidReturnType().isEmpty().toBlocking().single()).isTrue();
+        assertThat(dbProxyTestDao.insertVoidReturnType().hasElement().block()).isFalse();
```

Comment:
Isn't this a test of the Void type? Is it possible for a Mono<Void> to have an element?